src/freenet/node/useralerts/UserAlertManager.java

```diff
@@ -223,11 +225,35 @@ public HTMLNode renderAlert(UserAlert userAlert) {
 		userAlertNode.addChild("div", "class", "infobox-header", userAlert.getTitle());
 		HTMLNode alertContentNode = userAlertNode.addChild("div", "class", "infobox-content");
 		alertContentNode.addChild(userAlert.getHTMLText());
+		if (userAlert.canReply()) {
+			alertContentNode.addChild(renderReplyButton(userAlert, userAlert.getSourceNode()));
+		}
 		alertContentNode.addChild(renderDismissButton(userAlert, null));
 
 		return userAlertNode;
 	}
 
+	public HTMLNode renderReplyButton(UserAlert userAlert, WeakReference<PeerNode> peerNode) {
+		HTMLNode form = new HTMLNode("form",
+					new String[]{"method", "action"},
+					new String[]{"post", "/send_n2ntm/"});
+			form.addChild("input",
+					new String[]{"hidden", "name", "value"},
+					new String[]{"true", "replyTo", userAlert.getMessageText()
+							.replaceAll("\n", "NEWLINE")});
```

Comment:
Technically this could work, but it does feel a bit hacky.

That said, even without the replacement it should technically still be valid HTML5 (with the form field having the String value `"Foo\n\tBar"` in the example below):
```html
<input type="hidden" value="Foo
	Bar">
```

Unfortunately, the HTMLNode indentation might mess up the message.

More suitable approaches might be:
* Encoding the entire message in some reasonable encoding that does not produce newlines, like Base64.
* Referencing the original message itself (by its identifier) as an additional parameter, and retrieving the message on-the-fly when rendering the compose message form.

Reply:
I’m now encoding as base64. Since this is hidden anyway, that’s cleaner.